### PR TITLE
Refactor code, add unit and integration tests, fix config and update documentation.

### PR DESCRIPTION
**Pull request by tofran**
Hello, after having a hard time reading the code and not understanding what was going on, I decided to make it clearer, specifically in the  `SharedKey` module.

Most functionalities were not tested and there were some small issues, specially in the configuration. Hopefully this will bring more stability and make it _production ready_.

- [x] Fixed configuration where conection string was ignored :beetle:;
- [x] Fixed bug where query parameters were not sorted :bug:;
- [x] SharedKey: Renamed `formatted` to `format_date`;
- [x] General end-to-end tests for sign;
- [x] Query parameter handling (I think this was what was causing the most confusion): made it more idiomatic and is now being handled separated from the path;
- [x] Made the code always call the `get_url` for creating urls (in some places was being created by hand and others calling this func). Updated to make it responsible for all urls in `Azurex.Blob`;
- [x] Improved configuration errors;
- [x] Created full coverage tests for the configuration;
- [x] Created simple integration tests, to check that the lib actually works;
- [x] Changed Readme from "Currently only supports" to "Currently supports" 💪 ;
- [x] Added sample configuration to the README;
- [x] Update changelog  and bump versions;

Please leave your feedback,
Thank you

(Edits allowed by maintainers)

### CODE REVIEW

**Comment by jakobht** [comment]
Looks great - i see you are still working on it - please let me know when it's ready for review

**Comment by tofran** [comment]
Since this goes directly to master I decided to make everything that I had in my wish list.

Made some changes and lots of tests (updated in the top comment).
Integration tests can be run with `mix test --only=integration`, you can add that to the pipeline, but obviously it needs credentials to work.

What do you think?

**Comment by tofran** [comment]
image

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Fixed handling of multiple query parameters;
 - Fixed config `storage_account_connection_string` not doing anything.
 
+### Potential Breaking
+
+- Renamed `Azurex.Blob.get_blob_url(name, container \\ nil)` to `Azurex.Blob.get_url(container, name)`
```
Comment (tofran) [inline CHANGELOG.md:28]: Nicely spotted. If you wish we could also keep a backwards compatible function that calls the new one?

**Comment by jakobht** [comment]
Looks great - if you have time could you check out my few changes, and if you agree they are good I'll merge. Thanks a lot I really like the change for the URL parameters, that is really great. 
And its great to have some tests! :)

File: CHANGELOG.md (hunk 1 above)
Comment (jakobht) [reply CHANGELOG.md:28]: I think it's fine we are still at version  <1 so it's better to keep it clean

**Comment by tofran** [comment]
@jakobht perfect, I really like it. Specially the missing parameter from the connection string that I forgot.
:muscle: :shipit:

**Comment by jakobht** [comment]
Great - will merge then.

**Review by jakobht — APPROVED** [review]

**Pull request merged**